### PR TITLE
Fix debug assert with operation blocks in invalid code

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/74315

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@333fred ptal

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler One line change.  Tnx.

**Review by 333fred — APPROVED** [review]

File: src/Compilers/CSharp/Test/Emit2/Diagnostics/DiagnosticAnalyzerTests.cs [hunk 1]
```diff
@@ -4360,5 +4360,40 @@ internal async Task TestMinimumReportedSeverity(SeverityFilter severityFilter, D
             Assert.True(analyzer.AnalyzerInvoked);
             Assert.Equal(expectedMinimumReportedSeverity, analyzer.MinimumReportedSeverity);
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74315")]
+        public async Task TestOperationConstructorBlockCallbackOnInvalidBaseCall()
+        {
+            // lang=C#-Test
```
Comment (jjonescz) [inline src/Compilers/CSharp/Test/Emit2/Diagnostics/DiagnosticAnalyzerTests.cs:4367]: `lang=C#` turns on syntax highlighting in the string, right? but what does `lang=C#-Test` do?

**Review by jjonescz — APPROVED** [review]

File: src/Compilers/CSharp/Test/Emit2/Diagnostics/DiagnosticAnalyzerTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Emit2/Diagnostics/DiagnosticAnalyzerTests.cs:4367]: marks it as a C# test file, with supported C# test file syntax.  Note; we don't have `lang=c#` support (yet).  but it's something we could consider.

**Pull request merged**